lib/cdo/aws/cloudfront.rb

```diff
@@ -126,6 +132,43 @@ def self.create_or_update
       end
     end
 
+    # Returns the distribution ID and fetched DistributionConfig object for a given hostname.
+    # Tries the cached id-to-hostname mappings first if available.
+    # Forces reload if cached-id config is not found.
+    # Ignore cache if force=true.
+    def self.get_distribution_config(cloudfront, hostname, force=false)
+      id = get_distribution_id(cloudfront, hostname, force)
```

Comment:
Kind of aesthetics here - but if you can write this non-recursively, it'll be a lot more readable